export_saved.py

```diff
@@ -86,105 +87,179 @@ def get_args(argv):
-
-    Returns:
-        Reddit object instance.
-    """
-    username = None
-    password = None
-    client_id = None
-    client_secret = None
-    if args.username and args.password and args.client_id and args.client_secret:
-        username = args.username
-        password = args.password
-        client_id = args.client_id
-        client_secret = args.client_secret
-    else:
-        try:
-            import AccountDetails
-            username = AccountDetails.REDDIT_USERNAME
-            password = AccountDetails.REDDIT_PASSWORD
-            client_id = AccountDetails.CLIENT_ID
-            client_secret = AccountDetails.CLIENT_SECRET
-        except ImportError:
+class Reddit():
+    """A reddit instance to get back data"""
+    def __init__(self, args):
+        """init method.
+
+        Args:
+            args (argparse.Namespace): Parsed arguments.
+        """
+        # login
+        account = Reddit.login(args=args)
+        self.client_id = account['client_id']
+        self.client_secret = account['client_secret']
+        self.username = account['username']
+        self.password = account['password']
+        reddit = praw.Reddit(client_id=self.client_id,
+                             client_secret=self.client_secret,
+                             user_agent='export saved 2.0',
+                             username=self.username,
+                             password=self.password)
+        logging.debug('Login succesful')
+        self.reddit=reddit
+
+    @staticmethod
+    def login(args):
+        """login.
+
+        Args:
+            args (argparse.Namespace): Parsed arguments.
+
+        Returns:
+            Account details
+        """
+        username = None
+        password = None
+        client_id = None
+        client_secret = None
+        if args.username and args.password and args.client_id and args.client_secret:
+            username = args.username
+            password = args.password
+            client_id = args.client_id
+            client_secret = args.client_secret
+        else:
+            try:
+                import AccountDetails
+                username = AccountDetails.REDDIT_USERNAME
+                password = AccountDetails.REDDIT_PASSWORD
+                client_id = AccountDetails.CLIENT_ID
+                client_secret = AccountDetails.CLIENT_SECRET
+            except ImportError:
+                print(
+                    'You must specify a username, password, client id, '
+                    'and client secret, either in an AccountDetails file '
+                    'or by passing them as arguments (run the script with '
+                    'the --help flag for more info).'
+                )
+                exit(1)
+
+        if not username or not password or not client_id or not client_secret:
+            print('You must specify ALL the arguments')
             print(
-                'You must specify a username, password, client id, '
```

Comment:
No, sorry about that, this looks like a rebase that went wrong, that was not intentional